libs/design/src/atoms/form/checkbox/cva/checkbox-cva.directive.spec.ts

```diff
@@ -0,0 +1,93 @@
+import { Component } from '@angular/core';
+import { ComponentFixture, async, TestBed } from '@angular/core/testing';
+import { ReactiveFormsModule, FormControl } from '@angular/forms';
+import { By } from '@angular/platform-browser';
+
+import { DaffCheckboxComponent } from '../checkbox.component';
+import { DaffCheckboxModule } from '../checkbox.module';
+
+
+
+@Component({
+  template: `
+    <daff-checkbox name='test' value='testValue' [formControl]='checkbox'></daff-checkbox>
+  `
+})
+class CheckboxWrapperComponent {
+  checkbox = new FormControl()
+}
+
+describe('DaffCheckboxControlValueAccessorDirective', () => {
```

Comment:
I find these tests illegible. I really don't know what this is supposed to test.

"DaffCheckboxControlValueAccessorDirective with the directive?"

Reply:
So the describes aren't very clear so that definitely needs to be addressed. The way I designed this spec was to test that when using the directive it gave the component the CVA functionality. So I split the test into two different parts. One being `with the directive` and one being `without the directive`. With the directive I asserted that the CVA functions were being called. Then in `without the directive`, I didn't declare the directive inside the testing module and then I asserted that it would throw an error without the directive giving the component the CVA implementation.